interface/src/app/gene/search/search.js

```diff
@@ -264,18 +264,21 @@ angular.module('adage.gene.search', [
           for (var gene in $scope.selectedGenes) {
```

Comment:
I can think of two solutions:

(1) Use **hasOwnProperty()** method to filter out the inherited enumerable properties. (This solution is recommended by Douglas Crockford, the author of **jslint** and the popular book "JavaScript: the good parts"). Besides the possible performance issue, another downside of this solution is that even **hasOwnProperty()** method may be overwritten by a user and thus not do what is expected.

(2) Another solution is to create a dummy object without any prototype:
```dummyObj = Object.create(null);```
then add user-defined properties to this dummy object. You will be sure that this dummy object does not inherit any properties from anywhere.
The downside of this solution is that in many cases, you may have to add new properties to an object that already exists, thus you don't have the luxury to work from your dummy object.

I prefer the first solution.